### PR TITLE
Add statistics box

### PR DESCRIPTION
**Pull request by schwjm**
## What does this pull request change?
* Adds some fun information in the tracker, inspired mostly by me wanting to be able to automatically countdown how many places were left to check and how many more items were required for Defeat Ganondorf at a quick glance. The spirit of math inspired me to add some probability related stats as well.
* ~~Slightly improves the colors of the "other locations chest counts" by adding some shadowing.~~
* Fixes one errant HTML redundancy

## To-Do
- [x] cut "Chance of Endgame Find"
- [x] cut "Average Treasure Checks Remaining" from display
- [x] finalize stat names
- [x] move stat panel to squeeze in underneath other locations panel
- [x] add "Locations Checked"
- [x] rename / move "Treasure Locations Remaining"
- [x] respect "Hide Location Logic" by hiding stats that are derived from numbers/information hidden in this state (everything except "Locations Checked/Remaining"?)

## Testing
- [x] checked Firefox and Chrome that the launcher opened a window of sufficient size to not have scroll bars and show the new content just fine
- [x] sanity checking on the Locations Remaining math (e.g. locations remaining at launch matches what randomizer claimed at launch). notably, Defeat Ganondorf has to be exempted from the count
- [x] verified items remaining to finish game count is correct and counts down as you mark off triforce, progress sword/bow, etc.
- [x] sanity checked the endgame find probability formula (100% if only endgame items remain)
- [x] sanity checked average treasure checks remaining (= locations remaining if only endgame items remain)
- [x] tested that Hide Locaiton Logic hides all stats except Locations Checked/Remaining
- [x] tested wrapping with new float strategy and verified the stat box fits underneath other locations

## Other thoughts
Naming of the stats is obviously very open to suggestion, as well as what is included (or not) in the long run

### CODE REVIEW

**Comment by wooferzfg** [comment]
- The drop shadows don't look very good and should probably be excluded from this PR, at least for now.
- "Chance of Endgame Find" is confusing. I'm not sure that I understand exactly what it is.
- "Average Treasure Checks Remaining" and "Estimated Locations Leftover at End" make sense to some extent, but it's weird to have these when they're not based on the logic and how the randomizer actually generates seeds. I'm not sure we want to display statistics that aren't fully accurate.
- With 4 different "panels", we should probably make the default tracker size less wide but slightly taller so that it can fit 2 rows.

**Comment by schwjm** [comment]
> The first two stats are super useful and intuitive. We definitely want to keep those. Maybe add "Locations Checked" as well?
> "Treasure Locations Remaining" should maybe be renamed to just "Locations Remaining".

Sure. I'm thining "Locations Checked (Remaining)" and then display the countup and countdown on one line. Checked / Total or Remaining / Total might be good alternates.

> Hide the item countdown stat if location logic is hidden?

Makes sense.

> The drop shadows don't look very good and should probably be excluded from this PR, at least for now.

That's fine. It's completely unrelated to the rest and I don't think it's perfect either. I'm just going to rebase it out.

> "Chance of Endgame Find" is confusing. I'm not sure that I understand exactly what it is.

Basically the chance that your next item (irrespective of context) will be one needed for Defeat Ganondorf. In retrospect, I don't think it's super useful (it's just division of two numbers already shown), so I'd be fine with cutting it just to reduce clutter and save space (regarding suggestions below.)

> "Average Treasure Checks Remaining" and "Estimated Locations Leftover at End" make sense to some extent, but it's weird to have these when they're not based on the logic and how the randomizer actually generates seeds. I'm not sure we want to display statistics that aren't fully accurate.

Since one is just subtraction from the other, having both might just be noise. The thought about them is that they convey some sense of relative luck. For example if I start at "12 Estimated Leftover" but finish with 14 remaining then I had a somewhat lucky run where I checked fewer locations than average for that setting (plus this number will update as you go so you can see if you're "ahead" or "behind").

Not all locations are equal, of course, and the tracker has little or no context to decide what location you'll check next (any stats based on a location chosen should be shown in the location/map panel instead, I believe). Perhaps this metric can be improved, but the cases in which it could meaningfully apply logic would be rare or edgy. Consider that although extremely improbable, you could get all 19 endgame items as your first 19 checks--logic does not prohibit this, it only prohibits scenarios that are unwinnable. The opposite scenario, the only time this calculation would be misleading/fibbing a bit is if you were in a situation where non-endgame items must be found first, which I think would only happen if you had checked everywhere but dungeons with key lunacy off or some of the more heavily gated locations--even then I'm not sure the number would be off a ton. If there is a good way to account for this then I'm for implementing it, but I didn't see a good path to be aware of this. You'd have to be able to calculate "number of items before your next draw is impossible to be endgame" and then iterate from there.

**tl;dr**: I propose keeping "Estimated Locations Leftover at End" (better name can still be found) because it's somewhat more intuitive (higher = more relative luck) and the other one is usually very close to "locations remaining" anyway.

> With 4 different "panels", we should probably make the default tracker size less wide but slightly taller so that it can fit 2 rows. Another alternative is to just put the new panel directly below the extra locations one. The easiest way to do this would be to make a surrounding div that contains the extra locations and the stats.

I'll play with it a bit, I think the second suggestion will work well especially if some lean cuts are made to make sure everything fits in that gap nicely. The first suggestion is how I wanted to go originally but the map (2nd panel) having space reserved for the hover-over text made it awkward. If the window is shrunk then it will wrap like this but it didn't seem like a good default.

**Comment by wooferzfg** [comment]
These ideas seem pretty solid to me.

**Comment by wooferzfg** [comment]
- I think "Leftover" should be changed to "Left Over".
- "Show Single Color Background" should change the stats panel's background to be opaque.
- The buttons should always align to the bottom of the window like they used to.

Other than that, looks good!

**Comment by wooferzfg** [comment]
also, a side note: a lot of the "findElementById" calls in the code I wrote should probably be replaced at some point with jQuery stuff similar to what you did
